Core/Object Arts/Dolphin/MVP/Base/Dolphin MVP Base.pax

```diff
@@ -1275,6 +1283,23 @@ icon
 	^Icon fromId: 2 in: (ExternalResourceLibrary open: 'Shell32')! !
 !File class categoriesFor: #icon!constants!development!public! !
 
+!InputState methodsFor!
+
+userSessionIdleMs
+"
+	(Delay forSeconds: 1) wait. SessionManager inputState userSessionIdleMs.
+"
+	| currentTime flag lastInputInfo lastInputTime |
+	lastInputInfo := LASTINPUTINFO new.
+	flag := UserLibrary default getLastInputInfo: lastInputInfo.
+	flag ifFalse: [self error: 'Error obtaining last input time'].
+	currentTime := self millisecondClockValue.
```

Comment:
From MSDN it seems that `GetLastInputInfo` returns a millisecond clock value from `GetTickCount`. Dolphin's millisecond clock primitive is _not_ based on this timer. It's not based on the WinMM timer any more either (the comment `InputState>>millisecondClockValue` is wrong). In fact it uses the high precision CPU clock via the `QueryPerformanceCounter` API, converted to a millisecond value. This may or may not produce a value that has the same baseline as `GetTickCount`; testing on my machine just now, the QPC derived clock reported by the primitive seems to be about 1.2 seconds ahead of the millisecond clock reported by a call to `GetTickCount`. 
In conclusion I think you need to add `KernelLibrary>>getTickCount` and then use that instead to initialize `currentTime`.
```
!KernelLibrary methodsFor!

getTickCount
	<stdcall: dword GetTickCount>
	^self invalidCall! !
!KernelLibrary categoriesFor: #getTickCount!public!win32 functions-date and time! !
```